stdlib/Profile/src/Profile.jl

```diff
@@ -276,7 +276,7 @@ end
 function get_thread_ids(data::Vector{<:Unsigned}, taskid = nothing)
     threadids = Int[]
     for i in length(data):-1:1
-        if data[i] == 0 # find start of block
+        if i > 1 && is_block_end(data, i)
```

Comment:
This change and the change above are the actual fix. 
The rest is just removing code duplication